File: AndroidStudioStudy/5-AndroidDevelopmentArt/1/1.1/111/app/src/main/java/com/xiaoyingbo/a111/MainActivity.java
package com.xiaoyingbo.a111;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.xiaoyingbo.a111.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "demoMainActivity";

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding=ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(R.layout.activity_main);
        Log.d(TAG,"onCreate");

        Button btn = (Button)findViewById(R.id.btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this,SecondActivity.class));
                //finish();
            }
        });

        if(savedInstanceState!=null){
            String test = savedInstanceState.getString("test_text", "");
            Log.d(TAG,"onCreate||restore test_text:"+test);
        }
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        Log.d(TAG,"onSaveInstanceState");
        outState.putString("test_text","测试");
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);

        Log.d(TAG,"onRestoreInstanceState");
        String test = savedInstanceState.getString("test_text", "");
        if(!TextUtils.isEmpty(test)){
            Log.d(TAG,"onRestoreInstanceState||restore test_text:"+test);
        }
    }

    @Override
    public void onConfigurationChanged(@NonNull Configuration newConfig) {
        super.onConfigurationChanged(newConfig);

        Log.d(TAG,"onConfigurationChanged");
        Log.d(TAG,"onConfigurationChanged|newOrientation: "+newConfig.orientation);
    }

    @Override
    protected void onStop() {
        super.onStop();

        Log.d(TAG,"onStop");
    }

    @Override
    protected void onPause() {
        super.onPause();

        Log.d(TAG,"onPause");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        Log.d(TAG,"onDestroy");
    }

    @Override
    protected void onRestart() {
        super.onRestart();

        Log.d(TAG,"onRestart");
    }

    @Override
    protected void onResume() {
        super.onResume();

        Log.d(TAG,"onResume");
    }

    @Override
    protected void onStart() {
        super.onStart();

        Log.d(TAG,"onStart");
    }
}

File: Internship-Android-Markdown/view-learn/AnimatorDemo/senceaniamtiondemo/src/main/java/com/example/senceaniamtiondemo/MainActivity.kt
package com.example.senceaniamtiondemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ViewGroup
import android.widget.Button
import androidx.transition.Scene
import androidx.transition.Transition
import androidx.transition.TransitionInflater
import androidx.transition.TransitionManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val sceneRoot: ViewGroup = findViewById(R.id.scene_root)
        val aScene: Scene = Scene.getSceneForLayout(sceneRoot, R.layout.a_scene, this)
        val anotherScene: Scene = Scene.getSceneForLayout(sceneRoot, R.layout.another_scene, this)

        val inflateTransition =
            TransitionInflater.from(this)
                .inflateTransition(R.transition.fade_transition)

        val btn = findViewById<Button>(R.id.btn)
        var flag=true
        btn.setOnClickListener{
            flag = if(flag){
                TransitionManager.go(anotherScene,inflateTransition)
                false
            }else{
                TransitionManager.go(aScene,inflateTransition)
                true
            }
        }
    }
}